### PR TITLE
cmd/tailscaled, ipn/conffile: support ec2 user-data config file

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #1412
Updates #1866

### CODE REVIEW

File: cmd/tailscaled/tailscaled.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func main() {
 	flag.StringVar(&args.birdSocketPath, "bird-socket", "", "path of the bird unix socket")
 	flag.BoolVar(&printVersion, "version", false, "print version information and exit")
 	flag.BoolVar(&args.disableLogs, "no-logs-no-support", false, "disable log uploads; this also disables any technical support")
-	flag.StringVar(&args.confFile, "config", "", "path to config file")
+	flag.StringVar(&args.confFile, "config", "", "path to config file, or 'vm:user-data' to use the VM's user-data (EC2)")
```
Comment (maisem) [inline cmd/tailscaled/tailscaled.go:172]: On GCP it's common to have arbitrary metadata fields but I think aws only supports user-data.   Maybe we'll need to extend this in the future.

File: ipn/conffile/cloudconf.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	}
+	all, err := io.ReadAll(res.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read metadata token: %w", err)
+	}
+	return strings.TrimSpace(string(all)), nil
+}
+
+func readVMUserData() ([]byte, error) {
+	if omit.AWS {
```
Comment (maisem) [inline ipn/conffile/cloudconf.go:38]: Would be nice if this worked in more places, leave a TODO?

Comment (maisem) [reply ipn/conffile/cloudconf.go:38]: Like gcp/proxmox/etc

**Review by maisem — APPROVED** [review]

File: ipn/conffile/cloudconf.go (hunk 2 above)
Comment (bradfitz) [reply ipn/conffile/cloudconf.go:38]: done

File: ipn/conffile/cloudconf.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+	"tailscale.com/omit"
+)
+
+func getEC2MetadataToken() (string, error) {
+	if omit.AWS {
+		return "", omit.Err
+	}
+	req, _ := http.NewRequest("PUT", "http://169.254.169.254/latest/api/token", nil)
+	req.Header.Add("X-aws-ec2-metadata-token-ttl-seconds", "21600")
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply ipn/conffile/cloudconf.go:21]: made smaller

**Pull request merged**